### PR TITLE
add include and exclude patterns to jte-model (#243 Part 2)

### PR DESCRIPTION
**Pull request by chkl**
As suggested in #243

### CODE REVIEW

**Comment by casid** [comment]
Looks good to me! @edward3h, you know the model extension a lot better, I'd say you should have the final say before merge :-)

As with the other PR, maybe we could add a test for this, possible integration test? There are `test/jte-runtime-cp-test-models`(for Maven) and `test/jte-runtime-cp-test-models-gradle` (for Gradle), which could configured with such patterns?

File: jte-models/src/main/java/gg/jte/models/generator/ModelExtension.java [hunk 1]
```diff
@@ -31,6 +33,15 @@ public JteExtension init(Map<String, String> value) {
     @Override
     public Collection<Path> generate(JteConfig config, Set<TemplateDescription> templateDescriptions) {
         TemplateEngine engine = TemplateEngine.createPrecompiled(ContentType.Plain);
+
+        Pattern includePattern = modelConfig.includePattern();
+        Pattern excludePattern = modelConfig.excludePattern();
+
+        var templateDescriptionsFiltered = templateDescriptions.stream() //
+                .filter(x -> includePattern == null || includePattern.matcher(x.fullyQualifiedClassName()).matches()) //
+                .filter(x -> excludePattern == null || excludePattern.matcher(x.fullyQualifiedClassName()).matches()) //
```
Comment (kelunik) [inline jte-models/src/main/java/gg/jte/models/generator/ModelExtension.java:42]:
```suggestion
                .filter(x -> excludePattern == null || ! excludePattern.matcher(x.fullyQualifiedClassName()).matches()) //
```

**Comment by kelunik** [comment]
Also, this PR includes both changes now. We should either remove them or close #244.

**Comment by chkl** [comment]
> Also, this PR includes both changes now. We should either remove them or close #244.

Thanks @kelunik for reviewing it and good catch. I added the missing negation and dropped the commits that are already included in #244 .


I'm going to add tests for this and #244 later.

**Pull request merged**